modules/EvseV2G/connection.cpp

```diff
@@ -50,6 +50,29 @@ mbedtls_ssl_cache_context cache;
 static const int v2g_cipher_suites[] = {MBEDTLS_TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA256,
                                         MBEDTLS_TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256, 0};
 
+/* list of allowed hashes which our TLS server supports; this list is
+ * copied from mbedtls' ssl_preset_default_hashes (ssl_tls.c) with one
+ * nitpick: the MBEDTLS_MD_SHA1 is enabled unconditionally since TLSv1.2
+ * uses it as fallback when the client don't send 'signature_algorithms'
+ * extension; this means that we require this algo and including it without
+ * define results in a compile error in cause we try to build against
+ * a library where it is disabled (by error).
+ */
+static const int v2g_ssl_allowed_hashes[] = {
+#if defined(MBEDTLS_SHA512_C)
+    MBEDTLS_MD_SHA512,
+#endif
+#if defined(MBEDTLS_SHA512_C) && !defined(MBEDTLS_SHA512_NO_SHA384)
```

Comment:
This could be embedded in the previous block, but that's probably a question of style and preference. Something like:

```
#if defined(MBEDTLS_SHA512_C)
    MBEDTLS_MD_SHA512,
#  if !defined(MBEDTLS_SHA512_NO_SHA384)
    MBEDTLS_MD_SHA384,
#  endif
#endif
```